classes/coc/xxc/StoryCompiler.as

```diff
@@ -0,0 +1,153 @@
+/**
+ * Coded by aimozg on 28.08.2017.
+ */
+package coc.xxc {
+import coc.script.Eval;
+import coc.xlogic.Compiler;
+import coc.xlogic.Statement;
+import coc.xlogic.StmtList;
+import coc.xxc.stmts.*;
+
+public class StoryCompiler extends Compiler {
+	private var _basedir:String;
+
+	public function get basedir():String {
+		return _basedir;
+	}
+	public function StoryCompiler(basedir:String) {
+		this._basedir = basedir;
+		if (basedir.charAt(basedir.length-1)!='/') this._basedir+='/';
+	}
+	private var stack:/*Story*/Array = [];
+	public function clone(basedir:String=""):StoryCompiler {
+		return new StoryCompiler(basedir?basedir:_basedir);
+	}
+
+	public function attach(story:Story):StoryCompiler {
+		if (stack.length>0) throw new Error("StoryCompiler.attach called mid-compilation");
+		while(story) {
+			stack.push(story);
+			story = story.parent;
+		}
+		return this;
+	}
+	public function detach(story:Story):StoryCompiler {
+		while(story) {
+			if (stack.length == 0) throw new Error("Inconsistent stack during detach");
+			if (stack.pop() != story) throw new Error("Inconsistent stack during detach");
+			story = story.parent;
+		}
+		return this;
+	}
+	override protected function compileTag(tag:String, x:XML):Statement {
+		switch (tag) {
+			case "b":
+			case "i":
+				var list:StmtList = new StmtList();
+				list.stmts.push(new TextStmt("<"+tag+">",0));
+				compileChildrenInto(x,list.stmts);
+				list.stmts.push(new TextStmt("</"+tag+">",0));
+				return list;
+			case "display":
+				return compileDisplay(x);
+			case "dynStats":
+				return compileDynStats(x);
+			case "encounter":
+				return compileEncounter(x);
+			case "include":
+				return includeFile(x.@path, x.@required != "false");
+			case "output":
+				return compileOutput(x);
+			case "lib":
+			case "macro":
+				return compileStory(x, true);
+			case "set":
+				return compileSet(x);
+			case "story":
+			case "string":
```

Comment:
Cases without a break will fall-trought, is this intended?

https://help.adobe.com/en_US/ActionScript/3.0_ProgrammingAS3/WS5b3ccc516d4fbf351e63e3d118a9b90204-7fce.html
https://help.adobe.com/en_US/as2/reference/flashlite/WS5b3ccc516d4fbf351e63e3d118ccf9c47f-7edc.html